services/pull/pull.go

```diff
@@ -25,9 +25,13 @@ import (
 	"code.gitea.io/gitea/modules/notification"
 	"code.gitea.io/gitea/modules/process"
 	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/sync"
 	issue_service "code.gitea.io/gitea/services/issue"
 )
 
+// TODO: use clustered lock (unique queue? or *abuse* cache)
```

Comment:
? cc @lunny @zeripath ...

Reply:
We haven't introduce a distributed transaction in Gitea. But we really need one.
unique queue is a temporarily method to handle that I think.